app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt

```diff
@@ -46,7 +49,13 @@ fun SignInScreen(navigationActions: NavigationActions, userViewModel: UserViewMo
       authService.rememberFirebaseAuthLauncher(
           onAuthComplete = { result ->
             user = result.user
-            // todo call setCurrentUser in the viewmodel : userViewModel
+            userViewModel.viewModelScope.launch {
+              val fetchedUser = userViewModel.getUserByUid(user!!.uid)
+              if (fetchedUser == null) {
+                val newuser = User(user!!.uid, user!!.displayName!!, user!!.email!!, 0, emptyList())
+                userViewModel.addUser(newuser)
+              }
+            }
```

Comment:
It might be good to encapsulate the logic inside of a function to minimize the amount of code in the UI. Because none of the values defined there will be used except the viewModel.

For instance either in the userViewModel or the bottom of the SignIn.kt file. 
A simple signature taking the user (and maybe teh viewModel depending on where you do it) could do the job.

Reply:
Done I let you check the changes and mark as resolved